g11n-ws/modules/md-data-api-bundleimpl/src/main/java/com/vmware/vip/messages/data/dao/impl/ProductDao.java

```diff
@@ -126,29 +126,35 @@ public String getVersionInfo(String productName, String version)
 		return result;
 
 	}
-
 	/**
-	 * Get all product names and corresponding versions
-	 *
-	 * @return a map with the product names as key and with version list as value
-	 * @throws BundleException
-	 */
-	public Map<String, String[]> getProductsAndVersions() throws BundleException {
-		Map<String, String[]> productsAndVersions = new HashMap<>();
-		String basePath = bundleConfig.getBasePathWithSeparator() + ConstantsFile.L10N_BUNDLES_PATH;
-		File fileBase = new File(basePath);
-		if (fileBase.exists() && fileBase.isDirectory()) {
-			File[] productFolders = fileBase.listFiles();
-			for (File fileProduct : productFolders) {
-				if (fileProduct.isDirectory() && !fileProduct.isHidden()) {
-					String productName = fileProduct.getName();
-					String[] versions = fileProduct.list();
-					productsAndVersions.put(productName, versions);
-				}
-			}
-		} else {
-			throw new BundleException("The base l10n dir is not existing, the missed dir is: " + basePath);
-		}
-		return productsAndVersions;
-	}
+     * Get one product corresponding versions
+     *
+     * @return a map with the product names as key and with version list as value
+     * @throws BundleException
+     */
+    @Override
+    public List<String> getVersionList(String productName) throws DataException {
+        // TODO Auto-generated method stub
+        String basePath = bundleConfig.getBasePathWithSeparator() + ConstantsFile.L10N_BUNDLES_PATH + productName
+                + File.separator;
+        File fileBase = new File(basePath);
+        if (fileBase.exists() && fileBase.isDirectory()) {
+            String[] versionFoldernames = fileBase.list(new FilenameFilter() {
```

Comment:
Maybe it's better to change to "versionFolderNames"

Reply:
Because we also have db store media and S3 store media. there doesn't have fold concept

Reply:
done